samples/posix/gettimeofday/README.rst

```diff
@@ -12,6 +12,14 @@ system startup, the current time is queried using the SNTP networking
 protocol, enabled by setting the :option:`CONFIG_NET_CONFIG_CLOCK_SNTP_INIT`
 and :option:`CONFIG_NET_CONFIG_SNTP_INIT_SERVER` options.
 
+Requirements
```

Comment:
tl;dr: this is a clue, but not adequate for newbies.  Since it's what's done for other samples it's at least meeting the standard of practice.

I looked at "Networking with host", selected [QEMU Ethernet] https://docs.zephyrproject.org/latest/guides/networking/qemu_eth_setup.html#networking-with-eth-qemu)
because it seemed simplest, and verified `samples/net/sockets/echo_server` worked using telnet to 192.0.2.1:4242
(oddly the [sample recommended use of `echo-client`](https://docs.zephyrproject.org/latest/samples/net/sockets/echo_server/README.html#running-echo-client-in-linux-host) produced a `SO_BINDTODEVICE: No such device`, so that sample isn't completely documented either).

The gettimeofday sample, of course, doesn't use that variant of host networking; it seems to *require* the SLIP interface ("legacy...use only when necessary").

Following that procedure almost worked, except that I'm not running an SNTP server on my development host so it couldn't set the time:

```
gettimeofday(): HI(tv_sec)=0, LO(tv_sec)=5, tv_usec=30000
[00:00:05.030,000] <err> net_config: Cannot set time using SNTP
gettimeofday(): HI(tv_sec)=0, LO(tv_sec)=6, tv_usec=40000
```

For completeness and to assist people who do not normally use networking this should specify that "QEMU using SLIP" should be used, and that the local host must run an SNTP server.

(Unless the Ethernet networking is supposed to work, in which case that's a different problem.)